### PR TITLE
Define core and hosted projects, along with their requirements

### PR DESCRIPTION
**Pull request by fitzgen**

### CODE REVIEW

File: TSC/core-and-hosted-projects.md
Comment (penzn) [inline TSC/core-and-hosted-projects.md:58]: Was this  meant to be filled in?

**Review by penzn — COMMENTED** [review]
There is an 'etc' in Core Project requirements, not sure if that was meant to be filled in.

**Comment by xwang98** [comment]
We would like to nominate wamr as a core project. Shall it be started after this PR is approved and merged?

File: TSC/core-and-hosted-projects.md
Comment (fitzgen) [reply TSC/core-and-hosted-projects.md:58]: This is a template that is meant to be filled in by the project maintainers applying for core/hosted project status.

**Comment by fitzgen** [comment]
@xwang98 

> We would like to nominate wamr as a core project. Shall it be started after this PR is approved and merged?

Once this PR is merged then, yes, projects can begin the process described within and submit an application.

File: TSC/core-and-hosted-projects.md
Comment (penzn) [inline TSC/core-and-hosted-projects.md:58]: Even this is part of a template, this list is clearly labeled as _Core Project Requirements_, that sounds like a something that is supposed to be met before applying. If there are 'etc's that might be needed in addition to mission statement and changelog, maybe that should be spelled out.

File: TSC/core-and-hosted-projects.md
Comment (penzn) [inline TSC/core-and-hosted-projects.md:34]: Similar issue with this 'etc' - it in a list clearly labeled 'requirements', even if it is in a template.

File: TSC/core-and-hosted-projects.md
Comment (penzn) [reply TSC/core-and-hosted-projects.md:58]: Does it mean they can define their own requirements? The list is clearly labeled that way.

**Review by penzn — COMMENTED** [review]
Not sure I understood @fitzgen comment - can maintainers define their own requirements (it appears that templates allow for extra items on those lists)?

File: TSC/core-and-hosted-projects.md
Comment (tschneidereit) [reply TSC/core-and-hosted-projects.md:34]: I agree that it'd be good to make this as easy to follow as possible.

In fact, ideally we'd restore and update the proposal template, and turn it into an issue template, so that people have a very easy to follow guide for how to apply

**Comment by fitzgen** [comment]
I've moved the enumeration of the actual project requirements into github issue templates. Each requirement has an associated `TODO` to be filled in. This should address the confusion around what needs to be filled in when proposing a new hosted project or the promotion of a core project.

File: TSC/core-and-hosted-projects.md
Comment (tschneidereit) [inline TSC/core-and-hosted-projects.md:21]:
```suggestion
The Bytecode Alliance Technical Steering Committee (TSC) will provide timely acknowledgement of the application. The TSC will then perform an in-depth review of the project, evaluate whether it meets the core project requirements, and decide based on consensus whether to promote the project. Next, the TSC will reply in the issue with the results of the review and whether the project has been promoted. If it has been promoted, then the project maintainers will [choose a delegate to represent the project on the TSC](https://github.com/bytecodealliance/governance/blob/d0310b2f4fab7d5bd076627b196dfc0ffe040ec1/TSC/charter.md#composition).

## Handling of Existing Projects

At the point of adoption of these requirements, all existing projects are implicitly treated as Hosted Projects. The TSC will over time work with project maintainers to ensure that they satisfy the requirements for Hosted Projects, and explicitly add their projects to the list of Hosted Projects.

Additionally, we encourage maintainers of existing projects to consider applying for Core Project status, if they see their project as a good fit and in compliance with the stricter requirements.
```

**Review by ricochet — APPROVED** [review]

**Comment by tschneidereit** [comment]
Hey @xwang98, since it's been quite a while, I wanted to follow up here and give you and update on where things stand.

As you can see, this PR still hasn't been merged, because we still have some small things to work through. In the meantime however, what you can do to get things ready in parallel is to look at the criteria for both hosted projects and core projects that this PR introduces, and check how well WAMR satisfies them. As laid out in the PR, that will be the basis for deciding whether a project can gain Core Project status. Part of the process will be to have the project maintainers provide explanations for how the project satisfies the criteria, or what kinds of plans are in place to get to that place in cases where a criteria isn't yet satisfied.

To be clear, there's no need for you to act right now, but if you want to get a head-start on the process, this would be a great way to do so!

**Comment by xwang98** [comment]
@tschneidereit Thank you for the reminder. We have checked the criteria list, and WAMR seems in good shape.

File: .github/ISSUE_TEMPLATE/hosted-project.md
Comment (tschneidereit) [inline .github/ISSUE_TEMPLATE/hosted-project.md:27]: This seems like it's missing content?

File: .github/ISSUE_TEMPLATE/hosted-project.md
Comment (tschneidereit) [inline .github/ISSUE_TEMPLATE/hosted-project.md:183]: Based on today's TSC discussion, here's a suggested addition:

```suggestion

The TSC may waive the above contributor base requirements under certain conditions. In particular, the TSC may decide to adopt crucial upstream dependencies of existing Bytecode Alliance projects that are otherwise effectively unmaintained or only have a single maintainer.
```

**Comment by fitzgen** [comment]
I'm pleased to share that the TSC is ready to move forward with these requirements and merge this pull request. I am extremely proud of the high bar we are setting for ourselves and I am confident that by meeting and sustaining this level of rigor we can achieve the Bytecode Alliance's mission.

I want to thank everyone who shared feedback with us both in this PR thread and via private channels. You've definitely helped improve, clarify, and refine these requirements and the documents describing them; the Bytecode Alliance is better off because of it. Thanks!

**Review by tschneidereit — APPROVED** [review]
This came out really nice! Thank you so much for working through this! ❤️

**Pull request merged**
